contract/models/contract_line.py

```diff
@@ -650,13 +653,11 @@ def _check_start_end_dates(self):
     def _compute_create_invoice_visibility(self):
         today = fields.Date.context_today(self)
         for rec in self:
-            if rec.date_start:
-                if today < rec.date_start:
-                    rec.create_invoice_visibility = False
-                else:
-                    rec.create_invoice_visibility = bool(
-                        rec.recurring_next_date
-                    )
+            if (not rec.display_type and
+                    rec.date_start and today >= rec.date_start):
+                rec.create_invoice_visibility = bool(rec.recurring_next_date)
+            else:
+                rec.create_invoice_visibility = False
```

Comment:
This has risen me a great question: how do you control which sections/notes are invoiced each time? AFAIK, you are adding them each time you invoice, but depending on the lines, and their order, some sections have no meaning in certain invoices. Please put a big TODO about this and add a point in the ROADMAP about this. We are not going to cover it because we are overpassed in the budget and it's not our customer use-case, but we have to note it.

Reply:
True, I'm going to add that to the roadmap.
And now that you mention that, I think this change does not make sense since the sections and notes alone don't have to influence whether or not the invoices are generated. Am I right?

Reply:
Well, you can keep it this way any way for now while the ROADMAP is not solved.

Reply:
Or maybe the contrary (`True`), if you base what to pass to invoice on this field.